packages/core/src/hooks/useClickOutside/index.ts

```diff
@@ -17,6 +19,12 @@ export default function useClickOutside({
         return;
       }
 
+      if (ignoreClasses && event.target instanceof HTMLElement) {
+        if (event.target.closest(ignoreClasses.join(','))) {
+          return;
+        }
+      }
```

Comment:
Nit: avoid nested if statement, the condition can be extracted to a variable to simplify the statement